src/main/java/org/opensearch/neuralsearch/query/HybridQueryScorer.java

```diff
@@ -53,7 +50,7 @@ public HybridQueryScorer(final Weight weight, final List<Scorer> subScorers) thr
     HybridQueryScorer(final Weight weight, final List<Scorer> subScorers, final ScoreMode scoreMode) throws IOException {
         super(weight);
         this.subScorers = Collections.unmodifiableList(subScorers);
-        subScores = new float[subScorers.size()];
+        this.numSubqueries = subScorers.size();
```

Comment:
why do we need this variable? Why can't we use  this.subScorers.size() wherever it is required?

Reply:
Mainly for saving on CPU cycles. That call to size() poped up in the flame graph as a contributor to the CPU cycles time, single digit percentage.